src/BoundaryConditions/flux_calculations.jl

```diff
@@ -0,0 +1,70 @@
+# General
+"""
+    get_boundary_flux(model, ::NoFluxCondition, _...)
+
+Construct zero boundary fluxes for a scalar or 2-element vector(x,y)
+"""
+
+function get_boundary_flux(model, ::NoFluxCondition, var::Fields.Field, Ym, Ya)
+    FT = eltype(Ym)
+    flux = Geometry.Cartesian3Vector(FT(0))
+end
+
+# function get_boundary_flux(
+#     model,
+#     ::NoFluxCondition,
+#     ::Geometry.Cartesian12Vector,
+#     Ym,
+#     Ya,
+# )
+#     FT = eltype(Ym)
+#     flux = Geometry.Cartesian12Vector(FT(0), FT(0))
+# end
+
+# Momentum-specific
+"""
+    get_boundary_flux(model, ::DragLawCondition, _...)
+
+Construct non-zero boundary fluxes using the bulk formula and constant or Mohnin-Obukhov-based drag coefficient, Cd
+"""
+
+function get_boundary_flux(model, bc::DragLawCondition, uv, Ym, Ya)
+    FT = eltype(Ym)
+    coefficients = eltype(bc.coefficients) == FT ? bc.coefficients :
+        bc.coefficients(Ym, Ya)
+
+    uv_1 = Operators.getidx(Ym.uv, Operators.Interior(), 1)
+    u_wind = LinearAlgebra.norm(uv_1)
+
+    flux = Geometry.Cartesian3Vector(coefficients.Cd * u_wind) ⊗ uv_1
+end
+
+
+# Potential temperature density-specific
+"""
+    get_boundary_flux(model, ::DragLawCondition, _...)
+
+    Construct non-zero boundary fluxes using the bulk formula and constant or Mohnin-Obukhov-based heat transfer coefficient, Ch
+"""
+
+function get_boundary_flux(
+    model,
+    bc::BulkFormulaCondition,
+    ρθ::Fields.Field,
+    Ym,
+    Ya,
+)
+    FT = eltype(Ym)
+    coefficients = eltype(bc.coefficients) == FT ? bc.coefficients :
+        bc.coefficients(Ym, Ya)
+    T_sfc = bc.T_sfc
+
+    ρ_1 = Operators.getidx(Ym.ρ, Operators.Interior(), 1)
+    ρθ_1 = Operators.getidx(Ym.ρθ, Operators.Interior(), 1)
+    uv_1 = Operators.getidx(Ym.uv, Operators.Interior(), 1)
+    u_wind = LinearAlgebra.norm(uv_1)
+
+    flux = Geometry.Cartesian3Vector(
+        coefficients.Ch * u_wind * ρ_1 * (T_sfc - ρθ_1 / ρ_1),
```

Comment:
If I understand the operators correctly, here it should be `(ρθ_1 / ρ_1 - T_sfc)`, or rather `(ρθ_1 / ρ_1 - θ_sfc)`? You can see this is consistent with how you define the momentum flux in Line 39.

Reply:
Ooh yes, indeed, thanks for spotting the sign. :) (p.s., θ_sfc = T_sfc in the case)

Reply:
Just curious, why is this the case? Is it assuming the surface pressure to be the same as MSLP?

Reply:
It is, for this simple test case. But I corrected the notation to be consistent, as you suggested. :)